store/statstore/stat_store.go

```diff
@@ -327,10 +337,26 @@ func (ss *StatStore) Get(start, end time.Time) []TimePoint {
 
 	var result []TimePoint
 
-	if ss.IsEmpty() || (start.After(end) && end.After(time.Time{})) {
+	if start.After(end) && end.After(time.Time{}) {
+		return result
+	}
+
+	// Generate a TimePoint for the lastPut, if within range
+	low := start.Equal(time.Time{}) || start.Before(ss.lastPut.stamp)
+	hi := end.Equal(time.Time{}) || !end.Before(ss.lastPut.stamp)
+	if ss.lastPut.stamp.After(time.Time{}) && low && hi {
```

Comment:
If the initial check is not performed, then an empty StatStore would return an array with one empty TimePoint. Instead, we would want to skip the lastPut field as well and return an empty array, which would represent that no Put operations have been performed in the StatStore as of yet